lms/djangoapps/course_api/blocks/permissions.py

```diff
@@ -39,15 +40,14 @@ def can_access_self_blocks(requesting_user: User, course_key: CourseKey) -> Acce
     )
     if user_is_enrolled_or_staff:
         return ACCESS_GRANTED
-    try:
-        return is_course_public(course_key)
-    except ValueError:
```

Comment:
Is the `not course_exists(...)` check the only thing that could `throw` here?

If so, the `if not exists` check below seems to be enough, but want to check.

Reply:
`get_course` is what was throwing the valueerror when the course was nonexistent.

the `if not .course_exists(...)` call is safe, and guarantees that `get_course` will not throw said error